Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm

```diff
@@ -1136,7 +1148,7 @@ - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)data
 
         resourceResponse.setDeprecatedNetworkLoadMetrics(WebCore::copyTimingData(taskMetrics, networkDataTask->networkLoadMetrics()));
 
-        networkDataTask->didReceiveResponse(WTFMove(resourceResponse), negotiatedLegacyTLS, privateRelayed, [completionHandler = makeBlockPtr(completionHandler), taskIdentifier](WebCore::PolicyAction policyAction) {
+        networkDataTask->didReceiveResponse(WTFMove(resourceResponse), negotiatedLegacyTLS, privateRelayed, WTFMove(proxyName), [completionHandler = makeBlockPtr(completionHandler), taskIdentifier](WebCore::PolicyAction policyAction) {
```

Comment:
I think it would be more elegant to call setProxyName here instead of passing a parameter.  We also don't need a default for blob and data URLs because that can already be seen by WKNavigationResponse.response.URL.scheme.